fendermint/actors/bucket/src/actor.rs

```diff
@@ -59,8 +58,9 @@ impl Actor {
         validate_metadata(&params.metadata)?;
 
         let sub = if let Some(object) = state.get(rt.store(), &key)? {
-            // If we have existing blob
-            if params.overwrite {
+            // If we have existing blob and it's not expired
+            let expired = object.expiry <= rt.curr_epoch();
+            if params.overwrite || expired {
```

Comment:
don't require overwrite flag if the object is expired. otherwise it would be confusing when looking at the output from listing objects... the key won't show there anymore.